test-data/unit/semanal-errors.test

```diff
@@ -1419,3 +1419,12 @@ def g() -> None:
                           # N: (Hint: Use "T" in function signature to bind "T" inside a function)
 [builtins fixtures/dict.pyi]
 [out]
+
+[case testParamSpec]
+from typing_extensions import ParamSpec
+
+TParams = ParamSpec('TParams')
+TP = ParamSpec('?')  # E: String argument 1 '?' to ParamSpec(...) does not match variable name 'TP'
+TP2: int = ParamSpec('TP2')  # E: Cannot declare the type of a parameter specification
```

Comment:
Optionally, you can add a test case to `test-data/unit/semanal-symtable.test` and/or `test-data/unit/semanal-types.test` (these are not super high signal test cases, but on the other hand they don't require a lot of maintenance).